File: src/com/philhanna/flashcards/Card.java
package com.philhanna.flashcards;

/**
 * A flash card having a question and an answer.
 */
public interface Card {

   /**
    * Returns the question text
    */
   public String getQuestion();

   /**
    * Returns the answer text
    */
   public String getAnswer();

   /**
    * Toggles between showing its answer and showing its question. The card must
    * remember its current state.
    */
   public void toggle();
}

File: src/flashcards/domain/card.py
# flashcards.domain.card

from abc import ABC, abstractmethod


class Card(ABC):

    @property
    @abstractmethod
    def question(self) -> str: ...

    @property
    @abstractmethod
    def answer(self) -> str: ...


class BasicCard(Card):

    def __init__(self, question: str, answer: str) -> None:
        self._question = question
        self._answer = answer

    @property
    def question(self) -> str:
        return self._question

    @property
    def answer(self) -> str:
        return self._answer
